av_err2str build failure regression on Linux

https://github.com/shadps4-emu/shadPS4/blob/9d6eee6acfaeac7b33c8e99fb336a243d64c060b/src/core/libraries/videodec/videodec2_impl.cpp#L221 uses the unmodified av_err2str which causes build failures

previously fixed in https://github.com/shadps4-emu/shadPS4/pull/815, the modified macro exists in https://github.com/shadps4-emu/shadPS4/blob/9d6eee6acfaeac7b33c8e99fb336a243d64c060b/src/core/libraries/avplayer/avplayer_source.cpp#L25-L34

copying over that macro to videodec2_impl.cpp fixes the builds.
likely should move the macro to a common source point to prevent this regression in the future.

Regressed again:

```
In file included from /build/source/src/common/assert.h:7,
                 from /build/source/src/core/libraries/ajm/ajm_mp3.cpp:4:
/build/source/src/core/libraries/ajm/ajm_mp3.cpp: In member function 'AVFrame* Libraries::Ajm::AjmMp3Decoder::ConvertAudioFrame(AVFrame*)':
/build/source/src/core/libraries/ajm/ajm_mp3.cpp:97:64: error: taking address of temporary array
   97 |         LOG_ERROR(Lib_AvPlayer, "Could not convert frame: {}", av_err2str(res));
      |                                                                ^~~~~~~~~~
/build/source/src/common/logging/log.h:66:32: note: in definition of macro 'LOG_ERROR'
   66 |                                __VA_ARGS__)
      |                                ^~~~~~~~~~~
make[2]: *** [CMakeFiles/shadps4.dir/build.make:1214: CMakeFiles/shadps4.dir/src/core/libraries/ajm/ajm_mp3.cpp.o] Error 1
```

Introduced in https://github.com/shadps4-emu/shadPS4/pull/1456:

https://github.com/shadps4-emu/shadPS4/blob/941a668f78845290b4f5b28f2246c6cf00de765e/src/core/libraries/ajm/ajm_mp3.cpp#L95-L97

--- a/src/common/support/avdec.h
+++ b/src/common/support/avdec.h
@@ -1,0 +1,17 @@
+// SPDX-FileCopyrightText: Copyright 2024 shadPS4 Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+// support header file for libav
+
+// The av_err2str macro in libavutil/error.h does not play nice with C++
+#ifdef av_err2str
+#undef av_err2str
+#include <string>
+av_always_inline std::string av_err2string(int errnum) {
+    char errbuf[AV_ERROR_MAX_STRING_SIZE];
+    return av_make_error_string(errbuf, AV_ERROR_MAX_STRING_SIZE, errnum);
+}
+#define av_err2str(err) av_err2string(err).c_str()
+#endif // av_err2str

--- a/src/core/libraries/ajm/ajm_mp3.cpp
+++ b/src/core/libraries/ajm/ajm_mp3.cpp
@@ -12,6 +12,8 @@ extern "C" {
 #include <libswresample/swresample.h>
 }
 
+#include "common/support/avdec.h"
+
 namespace Libraries::Ajm {
 
 // Following tables have been reversed from AJM library

--- a/src/core/libraries/avplayer/avplayer_source.cpp
+++ b/src/core/libraries/avplayer/avplayer_source.cpp
@@ -18,16 +18,7 @@ extern "C" {
 #include <libswscale/swscale.h>
 }
 
-// The av_err2str macro in libavutil/error.h does not play nice with C++
-#ifdef av_err2str
-#undef av_err2str
-#include <string>
-av_always_inline std::string av_err2string(int errnum) {
-    char errbuf[AV_ERROR_MAX_STRING_SIZE];
-    return av_make_error_string(errbuf, AV_ERROR_MAX_STRING_SIZE, errnum);
-}
-#define av_err2str(err) av_err2string(err).c_str()
-#endif // av_err2str
+#include "common/support/avdec.h"
 
 namespace Libraries::AvPlayer {
 

--- a/src/core/libraries/videodec/videodec2_impl.cpp
+++ b/src/core/libraries/videodec/videodec2_impl.cpp
@@ -7,16 +7,7 @@
 #include "common/logging/log.h"
 #include "core/libraries/videodec/videodec_error.h"
 
-// The av_err2str macro in libavutil/error.h does not play nice with C++
-#ifdef av_err2str
-#undef av_err2str
-#include <string>
-av_always_inline std::string av_err2string(int errnum) {
-    char errbuf[AV_ERROR_MAX_STRING_SIZE];
-    return av_make_error_string(errbuf, AV_ERROR_MAX_STRING_SIZE, errnum);
-}
-#define av_err2str(err) av_err2string(err).c_str()
-#endif // av_err2str
+#include "common/support/avdec.h"
 
 namespace Libraries::Vdec2 {
 

--- a/src/core/libraries/videodec/videodec_impl.cpp
+++ b/src/core/libraries/videodec/videodec_impl.cpp
@@ -8,16 +8,7 @@
 #include "common/logging/log.h"
 #include "core/libraries/videodec/videodec_error.h"
 
-// The av_err2str macro in libavutil/error.h does not play nice with C++
-#ifdef av_err2str
-#undef av_err2str
-#include <string>
-av_always_inline std::string av_err2string(int errnum) {
-    char errbuf[AV_ERROR_MAX_STRING_SIZE];
-    return av_make_error_string(errbuf, AV_ERROR_MAX_STRING_SIZE, errnum);
-}
-#define av_err2str(err) av_err2string(err).c_str()
-#endif // av_err2str
+#include "common/support/avdec.h"
 
 namespace Libraries::Videodec {
 

